vector/src/test/java/im/vector/app/features/settings/devices/v2/ParseDeviceUserAgentUseCaseTest.kt

```diff
@@ -0,0 +1,138 @@
+        "Mozilla/5.0 (Windows NT 6.0; rv:40.0) Gecko/20100101 Firefox/40.0",
+        "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/42.0.2311.135 Safari/537.36 Edge/12.246",
+        )
+private val AN_EXPECTED_RESULT_LIST_FOR_WEB = listOf(
+        DeviceUserAgent(DeviceType.WEB, "Chrome", "Macintosh", null, null),
+        DeviceUserAgent(DeviceType.WEB, "Chrome", "Windows NT 10.0", null, null),
+        DeviceUserAgent(DeviceType.WEB, "Firefox", "Macintosh", null, null),
+        DeviceUserAgent(DeviceType.WEB, "Safari", "Macintosh", null, null),
+        DeviceUserAgent(DeviceType.WEB, "Chrome", "Android 9", null, null),
+        DeviceUserAgent(DeviceType.WEB, "Safari", "iPad", null, null),
+        DeviceUserAgent(DeviceType.WEB, "Safari", "iPhone", null, null),
+        DeviceUserAgent(DeviceType.WEB, "Firefox", "Windows NT 6.0", null, null),
+        DeviceUserAgent(DeviceType.WEB, "Edge", "Windows NT 10.0", null, null),
+)
+
+private val AN_UNKNOWN_USER_AGENT_LIST = listOf(
+        "AppleTV11,1/11.1",
+        "Curl Client/1.0",
+)
+private val AN_UNKNOWN_USER_AGENT_EXPECTED_RESULT_LIST = listOf(
+        DeviceUserAgent(DeviceType.UNKNOWN, null, null, null, null),
+        DeviceUserAgent(DeviceType.UNKNOWN, null, null, null, null),
+)
+
+class ParseDeviceUserAgentUseCaseTest {
+
+    private val parseDeviceUserAgentUseCase = ParseDeviceUserAgentUseCase()
+
+    @Test
+    fun `given an Android user agent then it should be parsed as expected`() {
+        A_USER_AGENT_LIST_FOR_ANDROID.forEachIndexed { index, userAgent ->
+            parseDeviceUserAgentUseCase.execute(userAgent) shouldBeEqualTo AN_EXPECTED_RESULT_LIST_FOR_ANDROID[index]
+        }
+    }
+
+    @Test
+    fun `given an iOS user agent then it should be parsed as expected`() {
+        A_USER_AGENT_LIST_FOR_IOS.forEachIndexed { index, userAgent ->
+            parseDeviceUserAgentUseCase.execute(userAgent) shouldBeEqualTo AN_EXPECTED_RESULT_LIST_FOR_IOS[index]
+        }
+    }
+
+    @Test
+    fun `given a Desktop user agent then it should be parsed as expected`() {
+        A_USER_AGENT_LIST_FOR_DESKTOP.forEachIndexed { index, userAgent ->
+            parseDeviceUserAgentUseCase.execute(userAgent) shouldBeEqualTo AN_EXPECTED_RESULT_LIST_FOR_DESKTOP[index]
+        }
+    }
+
+    @Test
+    fun `given a Web user agent then it should be parsed as expected`() {
+        A_USER_AGENT_LIST_FOR_WEB.forEachIndexed { index, userAgent ->
+            parseDeviceUserAgentUseCase.execute(userAgent) shouldBeEqualTo AN_EXPECTED_RESULT_LIST_FOR_WEB[index]
+        }
+    }
+
+    @Test
+    fun `given an unknown user agent then it should be parsed as expected`() {
+        AN_UNKNOWN_USER_AGENT_LIST.forEachIndexed { index, userAgent ->
+            parseDeviceUserAgentUseCase.execute(userAgent) shouldBeEqualTo AN_UNKNOWN_USER_AGENT_EXPECTED_RESULT_LIST[index]
+        }
+    }
```

Comment:
It is easier to navigate to failed test case in Android Studio or see the which one is failing on CI by checking function name on logs. If we make this generic it would take some more time to detect which platform is failing.